ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+P3A.swift

```diff
@@ -13,6 +13,71 @@ import Shared
 import os.log
 
 extension BrowserViewController {
+  enum CreateTabActionLocation {
+    case toolbar
+    case tabTray
+  }
+
+  func recordCreateTabAction(location: CreateTabActionLocation?) {
+    var tabCreationTotalStorage = P3ATimedStorage<Int>.tabCreationTotalStorage
+    var tabCreationFromToolbarStorage = P3ATimedStorage<Int>.tabCreationFromToolbarStorage
+
+    if let location {
+      tabCreationTotalStorage.add(value: 1, to: Date())
+      if location == .toolbar {
+        tabCreationFromToolbarStorage.add(value: 1, to: Date())
+      }
+    }
+
+    let toolbarCount = tabCreationFromToolbarStorage.combinedValue
+    let total = tabCreationTotalStorage.combinedValue
+
+    if total == 0 {
+      return
+    }
+
+    UmaHistogramRecordValueToBucket(
+      "Brave.Core.NewTabMethods",
+      buckets: [
+        .r(0..<25),
+        .r(25..<50),
+        .r(50..<75),
+        .r(75...),
+      ],
+      value: Int((Double(toolbarCount) / Double(total)) * 100)
+    )
+  }
+
+  func recordURLBarSubmitLocationP3A(from tab: Tab?) {
+    var urlSubmissionTotalStorage = P3ATimedStorage<Int>.urlSubmissionTotalStorage
+    var urlSubmissionNewTabStorage = P3ATimedStorage<Int>.urlSubmissionNewTabStorage
+
+    if let tab, !tab.isPrivate {
```

Comment:
Not sure if we need this private mode check

Reply:
P3A cannot report on private mode usage, so it's best that we keep it

Reply:
Okay I should also guard `recordCreateTabAction` as well in private mode